src/main/scala/it/pagopa/interop/authorizationserver/service/impl/AuthorizationManagementServiceImpl.scala

```diff
@@ -22,9 +24,10 @@ class AuthorizationManagementServiceImpl(
 
   override def getKeyWithClient(clientId: UUID, kid: String)(implicit
     contexts: Seq[(String, String)]
-  ): Future[KeyWithClient] =
-    withHeaders { (_, correlationId, ip) =>
-      val request =
+  ): Future[KeyWithClient] = fastGetOpt(contexts)(CORRELATION_ID_HEADER)
```

Comment:
⚠️ the correlation id is not currently in the context (the user should not send it as parameter).
I agree though that here we should have one, we need to generate it in the `createToken` and pass it to all our internal calls

Reply:
How can it be? Don't we have it in the contexts in the endpoint that calls this api? I thought that it was created for every kind of call, being a M2M or a FE one. If that's the case, it won't get logged/created for the createToken and so it's kinda pointless to send a new one downstream, innit?

Reply:
In case of FE, is the FE itself that sends the correlation id.
Anyway my bad, I forgot that the `withLoggingAttributes` directive creates the correlation id if required 👍

Reply:
😥